### DISCUSSION TITLE
How to generate a 3D grid from a segy file?

### DISCUSSION THREAD

**Discussion by chenyi-maker** [krober10nd/SeismicMesh#238] [General]
I tried to generate a 3D grid from a segy file, but the program kept running with errors. Here is my program. Attached is the corresponding segy file. I find that using segy files is always np.ndim==2. In function read_segy, ny returns 0. How do I solve this problem? Brain3D.segy data is written by me through Matlab, which is an array of 480 *480 *350.
![Matlab](https://user-images.githubusercontent.com/115963126/219255367-e9768b2a-3863-4015-9844-d57692dbf25a.png)

![mistake](https://user-images.githubusercontent.com/115963126/219252735-9a73cac8-5738-411d-91e3-0e97d8faf316.png)
[Brain3D.zip](https://github.com/krober10nd/SeismicMesh/files/10751520/Brain3D.zip)

**Comment by krober10nd** [comment]
Hi, see the README example for an example of 3D EAGE meshing. Just save a 3d numpy array as a pickle. 

The major issue is that the seismic community has refused to use data standards (eg NetCDF) that nearly every other geophysical discipline has adopted. I could not understand the 3D segy format and thus never implemented a reader for that. 

In any case if you know how to read this file format, feel free to submit a pull request with this example. 

If your domain is irregularly shaped (eg a brain), you can generate a signed distance function to represent it rather than use the standard shapes. There is another discussion of someone doing that to model the irregular free surface in 3d. 

Best of luck

**Comment by chenyi-maker** [comment]
Thank you very much for reminding me. I went to see the part about 3D grid generation (Edge), and I used fread function of matlab to write 480 *480 *350 array data into Brain3D.bin in the form of binary (bin) file. File, but I still get an error when I run the program. The original data file is a regular cuboid (no irregular surface fluctuations).
![mistake1](https://user-images.githubusercontent.com/115963126/219368155-e375ec08-b646-47ab-ae74-e99b737623ef.png)
![mistake2](https://user-images.githubusercontent.com/115963126/219368163-336db882-5df5-4e67-b1c5-88e10c36b5ce.png)
[Brain3D.zip](https://github.com/krober10nd/SeismicMesh/files/10756111/Brain3D.zip)
